js/phemex.js

```diff
@@ -2794,6 +2807,79 @@ module.exports = class phemex extends Exchange {
         return await this.privatePutPositionsLeverage (this.extend (request, params));
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const accountsByType = this.safeValue (this.options, 'accountsByType', {});
+        const fromId = this.safeString (accountsByType, fromAccount);
+        const toId = this.safeString (accountsByType, toAccount);
+        let direction = undefined;
+        if (fromId === undefined) {
+            const keys = Object.keys (accountsByType);
+            throw new ExchangeError (this.id + ' fromAccount must be one of ' + keys.join (', '));
+        }
+        if (toId === undefined) {
+            const keys = Object.keys (accountsByType);
+            throw new ExchangeError (this.id + ' toAccount must be one of ' + keys.join (', '));
+        }
+        if (fromId === 'spot' && toId === 'future') {
+            direction = '2';
+        }
+        if (fromId === 'future' && toId === 'spot') {
+            direction = '1';
+        }
+        if (direction === undefined) {
+            throw new ExchangeError (this.id + ' transfer can only be down from future to spot or from spot to future');
+        }
+        const scaledAmmount = this.toEv (amount, currency);
+        const request = {
+            'currency': currency['id'],
+            'moveOp': direction,
+            'amountEv': scaledAmmount,
+        };
+        const response = await this.privatePostAssetsTransfer (this.extend (request, params));
+        //
```

Comment:
Let's move the parsing code in lines 2841-2870 to `parseTransfer` (see mentions of `parseTransfer` in the other exchanges).